pom.xml

```diff
@@ -95,131 +95,186 @@
+        <jboss.snapshots.repo.id>jboss-snapshots-repository</jboss.snapshots.repo.id>
+        <jboss.snapshots.repo.url>https://repository.jboss.org/nexus/content/repositories/snapshots/</jboss.snapshots.repo.url>
 
-        <maven-surefire-plugin.version>2.21.0</maven-surefire-plugin.version>
+        <!-- Build settings -->
 
-        <!-- Used to add further arguments to the arg line for specific sub-modules and profiles -->
-        <maven-surefire-plugin.argLine></maven-surefire-plugin.argLine>
-        <maven-surefire-plugin.argLine.add-modules></maven-surefire-plugin.argLine.add-modules>
-        <maven-surefire-plugin.argLine.add-opens></maven-surefire-plugin.argLine.add-opens>
+        <hibernate-validator-parent.path>.</hibernate-validator-parent.path>
+
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
+        <!-- No need to build the javadocs per module. Aggregated javadocs are build in the distribution module. See also HV-894 -->
+        <maven.javadoc.skip>true</maven.javadoc.skip>
+
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.testTarget>${maven.compiler.target}</maven.compiler.testTarget>
+        <maven.compiler.testSource>${maven.compiler.source}</maven.compiler.testSource>
         <!--
-        Names of modules to be passed via "add-modules" option when forking java through Surefire and javadoc
-        plugins on Java 9
+                Options to override the compiler arguments directly on the compiler argument line to separate between what
+                the IDE understands as the source level and what the Maven compiler actually use.
         -->
-        <maven-surefire-plugin.jigsaw.modules>java.xml.bind</maven-surefire-plugin.jigsaw.modules>
-        <maven-javadoc-plugin.jigsaw.modules>java.xml.bind</maven-javadoc-plugin.jigsaw.modules>
-
-        <!-- add-opens options required for Arquillian and WildFly -->
-        <arquillian.javaVmArguments.add-opens></arquillian.javaVmArguments.add-opens>
+        <maven.compiler.argument.target>${maven.compiler.target}</maven.compiler.argument.target>
+        <maven.compiler.argument.source>${maven.compiler.source}</maven.compiler.argument.source>
+        <maven.compiler.argument.testTarget>${maven.compiler.testTarget}</maven.compiler.argument.testTarget>
+        <maven.compiler.argument.testSource>${maven.compiler.testSource}</maven.compiler.argument.testSource>
 
-        <!-- Forbidden API related properties -->
-        <forbiddenapis-junit.path>forbidden-junit.txt</forbiddenapis-junit.path>
+        <!-- Set empty default values to avoid Maven leaving property references (${...}) when it doesn't find a value -->
+        <surefire.jvm.args></surefire.jvm.args>
```

Comment:
So, to explain the rationale... In Search, I used "jvm.args" for properties containing arguments that are specific to a version of the JVM, and "system.args" for the actual arguments passed to surefire/failsafe. Yeah, it's dumb.

If you don't use the exact same property names as Search, and thus are not constrained by consistency with Search, I would recommend to use clearer names, like what I proposed above.